wandb/sdk/data_types/video.py

```diff
@@ -155,29 +160,36 @@ def encode(self, fps: int = 4) -> None:
         )
         if TYPE_CHECKING:
             kwargs: Dict[str, Optional[bool]] = {}
-        try:  # older versions of moviepy do not support logger argument
-            kwargs = {"logger": None}
-            if self._format == "gif":
-                write_gif_with_image_io(clip, filename)
-            else:
-                clip.write_videofile(filename, **kwargs)
-        except TypeError:
-            try:  # even older versions of moviepy do not support progress_bar argument
-                kwargs = {"verbose": False, "progress_bar": False}
-                if self._format == "gif":
-                    clip.write_gif(filename, **kwargs)
-                else:
-                    clip.write_videofile(filename, **kwargs)
-            except TypeError:
-                kwargs = {
-                    "verbose": False,
-                }
-                if self._format == "gif":
-                    clip.write_gif(filename, **kwargs)
-                else:
-                    clip.write_videofile(filename, **kwargs)
+
+        kwargs = {"logger": None}
+        if self._format == "gif":
+            write_gif_with_image_io(clip, filename)
+        else:
+            clip.write_videofile(filename, **kwargs)
+
         self._set_file(filename, is_tmp=True)
 
+    async def _encode_with_spinner(self, fps: int = 4) -> None:
+        term_printer = printer.new_printer()
+        encoding_done = asyncio.Event()
+
+        async def update_spinner(encoding_done: asyncio.Event):
```

Comment:
Need to annotate the return type to make sure this is type-checked.

```suggestion
        async def update_spinner(encoding_done: asyncio.Event) -> None:
```

Also, you don't need to pass `encoding_done` as a parameter, since it's not reassigned inside the method body.